Source/Csla.AspNetCore/Blazor/State/StateController.cs

```diff
@@ -37,23 +38,33 @@ public class StateController(ApplicationContext applicationContext, ISessionMana
     /// Gets current user session data in a serialized
     /// format.
     /// </summary>
+    /// <param name="lastTouched">Last touched value from session</param>
     /// <returns></returns>
     [HttpGet]
-    public virtual byte[] Get()
+    public virtual StateResult Get(long lastTouched)
     {
+      var result = new StateResult();
       var session = _sessionManager.GetSession();
-      session.IsCheckedOut = true;
-      var message = (SessionMessage)ApplicationContext.CreateInstanceDI(typeof(SessionMessage));
-      message.Session = session;
-      if (FlowUserIdentityToWebAssembly)
+      if (session.LastTouched == lastTouched)
       {
-        var principal = new CslaClaimsPrincipal(ApplicationContext.Principal);
-        message.Principal = principal;
+        result.ResultStatus = ResultStatuses.NoUpdates;
       }
-      var formatter = new MobileFormatter(ApplicationContext);
-      var buffer = new MemoryStream();
-      formatter.Serialize(buffer, message);
-      return buffer.ToArray();
+      else
+      {
+        var message = (SessionMessage)ApplicationContext.CreateInstanceDI(typeof(SessionMessage));
+        message.Session = session;
+        if (FlowUserIdentityToWebAssembly)
+        {
+          var principal = new CslaClaimsPrincipal(ApplicationContext.Principal);
+          message.Principal = principal;
+        }
+        var formatter = Csla.Serialization.SerializationFormatterFactory.GetFormatter(ApplicationContext);
```

Comment:
My future aspirations in work here: Instead of getting it from this static class just let it be injected into the Controller as a field.